theano/tensor/blas.py

```diff
@@ -250,10 +270,11 @@ def perform(self, node, inputs, out_storage):
             out = numpy.dot(A, x)
             if alpha != 1:
                 out *= alpha
-            if beta != 1:
-                out += beta * y
-            else:
-                out += y
+            if beta != 0:
+                if beta != 1:
+                    out += beta * y
+                else:
+                    out += y
```

Comment:
This hides nan when beta=0 which is what we should do.  They will appear otherwise.